src/lib/repositories/impl_v2/oauth2_repository_impl.py

```diff
@@ -104,8 +104,9 @@ def validate_token(self, token):
         # That is the way how jwt library validates the token
         jwt.decode(token, self.env_config.oauth2_secret_key, algorithms="HS256")
 
-    def refresh_token(self, refresh_token):
+    def refresh_token(self, refresh_token, client_id, client_secret):
```

Comment:
Make sure the old access token is expired, if not return the same old access token since it is still valid

Reply:
done.